projects/example-app/src/app/books/containers/find-book-page.component.ts

```diff
@@ -13,9 +13,9 @@ import * as fromBooks from '@example-app/books/reducers';
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
     <bc-book-search
-      [query]="searchQuery$ | async"
-      [searching]="loading$ | async"
-      [error]="error$ | async"
+      [query]="(searchQuery$ | async)!"
```

Comment:
is it good practice to assert streams as truthy? I don't see why this is needed as `searchQuery$` is defined as `Observable<string>`, not `Observable<string | undefined>`.

if this behavior is because async can emit `undefined` first, I feel like there's another way to solve as I totally develop apps in strict mode with `async` and haven't run into needing the bang...

Reply:
otherwise IDE swears -
Type 'string | null 'is not assignable to type' string '.
   Type 'null' is not assignable to type 'string'

Reply:
what about
```suggestion
      [query]="(searchQuery$ | async) ?? ''"
```

in general, I think providing a value when the stream is falsy is better than using `!` to assert that it is not falsy because it avoids the issue that `async` really does output `null` as an initial value, which can cause problems in an input declares it only accepts `string`, for example.